test/unit/test_matrix.cpp

```diff
@@ -1153,3 +1153,47 @@ TEST(MatrixDestructorFutures, ConstAfterRead) {
 
   last_task.get();
 }
+
+TEST(MatrixExceptionPropagation, PropagateOnRwWithRWAccess) {
+  auto matrix = createMatrix<TypeParam>();
+
+  auto f = matrix(LocalTileIndex(0, 0)).then(hpx::util::unwrapping([](auto&&) {
+    throw std::runtime_error("exception in task");
+  }));
+
+  EXPECT_ANY_THROW(matrix(LocalTileIndex(0, 0)).get());
+  EXPECT_ANY_THROW(f.get());
+}
+
+TEST(MatrixExceptionPropagation, PropagateOnRwWithReadAccess) {
+  auto matrix = createMatrix<TypeParam>();
+
+  auto f = matrix(LocalTileIndex(0, 0)).then(hpx::util::unwrapping([](auto&&) {
+    throw std::runtime_error("exception in task");
+  }));
+
+  EXPECT_ANY_THROW(matrix.read(LocalTileIndex(0, 0)).get());
+  EXPECT_ANY_THROW(f.get());
```

Comment:
I would check that the type and message of the exception is correct. (Same for other cases)

Reply:
Currently we throw `std::runtime_error("An exception has been thrown and this tile destroyed")));`

What do you think of adding a custom exception for this?

Reply:
It can be a good option, since there is no ways to get the correct exception.